src/components/dock/dock-button/dock-button.spec.tsx

```diff
@@ -0,0 +1,35 @@
+import { h } from '@stencil/core';
+import { newSpecPage, SpecPage } from '@stencil/core/testing';
+import { DockItem } from '../dock.types';
+import { DockButton } from './dock-button';
+
+let page: SpecPage;
+let item: DockItem;
+let eventSpy: jest.Mock;
+
+beforeEach(async () => {
+    item = {
+        id: 'tables',
+        label: 'Tables',
+        icon: 'insert_table',
+        dockMenu: { componentName: 'my-custom-menu' },
+    };
+    page = await newSpecPage({
+        components: [DockButton],
+        template: () => <limel-dock-button item={item} />,
+    });
+    eventSpy = jest.fn();
+    page.body.addEventListener('dockMenuOpened', eventSpy);
```

Comment:
I would move this into the test. Creating a globally scoped `eventSpy` makes it less clear what event is spied on, and other tests (when added) might not need to spy on any event.